dashboard/app/models/datablock_storage_record.rb

```diff
@@ -20,8 +20,15 @@ class DatablockStorageRecord < ApplicationRecord
   # Composite primary key:
   self.primary_keys = :project_id, :table_name, :record_id
 
-  # TODO: #57001, implement enforcement of MAX_RECORD_LENGTH, we already have
-  # a test for this, but we're skipping it until this is implemented. This
-  # should ensure the string form of .json is less than 4096 bytes.
+  validate :max_record_length
+
+  StudentFacingError = DatablockStorageTable::StudentFacingError
+
   MAX_RECORD_LENGTH = 4096
+
+  private def max_record_length
+    if record_json.to_json.bytesize > MAX_RECORD_LENGTH
+      raise StudentFacingError.new(:MAX_RECORD_LENGTH_EXCEEDED), "JSON data cannot exceed #{MAX_RECORD_LENGTH} bytes"
```

Comment:
Will a student understand 'JSON data' in this context? Might need a message that aligns more with ui language? 'Table record size' or something?

Reply:
or take the wording we landed on before: "The record is too large. The maximum allowable size is #{DatablockStorageRecord::MAX_RECORD_LENGTH} bytes"

Reply:
ohhhh yeah, that was a good error message wasn't it 🤣